sdk/identity/identity/test/public/node/azurePipelinesCredential.spec.ts

```diff
@@ -41,14 +43,42 @@ describe("AzurePipelinesCredential", function () {
       tenantId,
       clientId,
       "existingServiceConnectionId",
-      systemAccessToken,
+      systemAccessToken
     );
     const regExp: RegExp =
-      /AzurePipelinesCredential: Authenticated Failed. Received null token from OIDC request. Response status- 404./;
+      /AuthenticationRequiredError: invalid_client: Error\(s\): 700213 .* AADSTS700213: No matching federated identity record found for presented assertion subject .* Please note that the matching is done using a case-sensitive comparison. Check your federated identity credential Subject, Audience and Issuer against the presented assertion/;
     await assert.isRejected(
       credential.getToken(scope),
       regExp,
-      "error thrown doesn't match or promise not rejected",
+      "error thrown doesn't match or promise not rejected"
+    );
+  });
+
+  it.only("failure includes the expected response headers", async function () {
+    if (!isLiveMode()) {
+      this.skip();
+    }
+    // clientId for above service connection
+    const clientId = process.env.AZURE_SERVICE_CONNECTION_CLIENT_ID!;
+    const systemAccessToken = process.env.SYSTEM_ACCESSTOKEN!;
+    const credential = new AzurePipelinesCredential(
+      tenantId,
+      clientId,
+      "existingServiceConnectionId",
+      systemAccessToken
+    );
+    const regExpHeader1: RegExp =
```

Comment:
This test is a little complex for what it does. Can we just check for strings?

await assert.isRejected(credential.getToken(scope), /x-msedge-ref/)

etc?

Reply:
Yeah I could simplify that though not sure if "-" and hexadecimal characters would be part of the "w". Actually more importantly, `assert.isRejected` doesn't check the multiline error description. It only does pattern matching for the error message :(

Reply:
You could "drop down" to your own implementation of the test like:

```ts
try {
   await asyncOperation();
   assert.fail("expected an error, but did not get one");
catch (e: any) {
   assert.includes(error.error_description, "my string to check for");
}
```

Etc... `isRejected` is a convenience method but probably doesn't do anything special so you can always use that pattern (and its the pattern we used to use before chai-as-promised was added). Does that help?